pkg/controllers/nodeclaim/consistency/controller.go

```diff
@@ -95,18 +99,46 @@ func (c *Controller) Reconcile(ctx context.Context, nodeClaim *v1beta1.NodeClaim
 	if err != nil {
 		return reconcile.Result{}, nodeclaimutil.IgnoreDuplicateNodeError(nodeclaimutil.IgnoreNodeNotFoundError(err))
 	}
+	if err = c.checkConsistency(ctx, nodeClaim, node); err != nil {
+		return reconcile.Result{}, err
+	}
+	if !equality.Semantic.DeepEqual(stored, nodeClaim) {
+		// We call Update() here rather than Patch() because patching a list with a JSON merge patch
+		// can cause races due to the fact that it fully replaces the list on a change
+		// Here, we are updating the status condition list
+		if err = c.kubeClient.Status().Update(ctx, nodeClaim); client.IgnoreNotFound(err) != nil {
+			if errors.IsConflict(err) {
+				return reconcile.Result{Requeue: true}, nil
+			}
+			return reconcile.Result{}, err
+		}
+	}
+	return reconcile.Result{RequeueAfter: scanPeriod}, nil
+}
+
+func (c *Controller) checkConsistency(ctx context.Context, nodeClaim *v1beta1.NodeClaim, node *v1.Node) error {
+	hasIssues := false
 	for _, check := range c.checks {
 		issues, err := check.Check(ctx, node, nodeClaim)
 		if err != nil {
-			return reconcile.Result{}, fmt.Errorf("checking node with %T, %w", check, err)
+			return fmt.Errorf("checking node with %T, %w", check, err)
 		}
 		for _, issue := range issues {
 			log.FromContext(ctx).Error(err, "consistency error")
 			consistencyErrors.With(prometheus.Labels{checkLabel: reflect.TypeOf(check).Elem().Name()}).Inc()
 			c.recorder.Publish(FailedConsistencyCheckEvent(nodeClaim, string(issue)))
 		}
+		hasIssues = hasIssues || (len(issues) > 0)
 	}
-	return reconcile.Result{RequeueAfter: scanPeriod}, nil
+	// If status condition for consistent state is not true but no issues are found clear the status condition
```

Comment:
comment needs update
```suggestion
	// If status condition for consistent state is not true and no issues are found, set the status condition to true
```